yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala

```diff
@@ -306,10 +305,9 @@ private object YarnClusterDriver extends Logging with Matchers {
       System.exit(1)
     }
 
-    listener = new SaveExecutorInfo
     val sc = new SparkContext(new SparkConf()
+      .set("spark.extraListeners", "org.apache.spark.deploy.yarn.SaveExecutorInfo")
```

Comment:
Man I really dislike this API, but it's what we have now...

nit: `classOf[SaveExecutorInfo].getName()` instead of the hardcoded string.